lte/gateway/c/sctpd/sctpd_downlink_impl.cpp

```diff
@@ -34,35 +36,51 @@ Status SctpdDownlinkImpl::Init(
   const InitReq *req,
   InitRes *res)
 {
-  MLOG(MDEBUG) << "SctpdDownlinkImpl::Init starting";
-
-  if (_sctp_connection != nullptr && !req->force_restart()) {
-    MLOG(MINFO) << "SctpdDownlinkImpl::Init reusing existing connection";
-    res->set_result(InitRes::INIT_OK);
-    return Status::OK;
-  }
-
-  if (_sctp_connection != nullptr) {
-    MLOG(MDEBUG)
-      << "SctpdDownlinkImpl::Init cleaning up sctp_desc and listener";
-
-    auto conn = std::move(_sctp_connection);
-    conn->Close();
-  }
-
-  MLOG(MDEBUG) << "SctpdDownlinkImpl::Init creating new socket and listener";
-
-  try {
-    _sctp_connection = std::make_unique<SctpConnection>(*req, _uplink_handler);
-  } catch (...) {
-    res->set_result(InitRes::INIT_FAIL);
-    return Status::OK;
-  }
-
-  _sctp_connection->Start();
-
-  res->set_result(InitRes::INIT_OK);
-  return Status::OK;
+  MLOG(MINFO) << "SctpdDownlinkImpl::req->ppid()=" << std::to_string(req->ppid());
+  MLOG(MINFO) << "SctpdDownlinkImpl::req->port()=" << std::to_string(req->port());
+  
+  if (req->ppid() == S1AP) {
+	   if (_sctp_4G_connection != nullptr && !req->force_restart()) {
+		    MLOG(MERROR) << "SctpdDownlinkImpl::Init reusing existing connection";
+		    res->set_result(InitRes::INIT_OK);
+		    return Status::OK;
+	   }
+	   if (_sctp_4G_connection != nullptr) {
+		    MLOG(MINFO)<< "SctpdDownlinkImpl::Init cleaning up sctp_desc and listener";
+		    auto conn = std::move(_sctp_4G_connection);
+		    conn->Close();
+	   }
+	   MLOG(MINFO) << "SctpdDownlinkImpl::Init creating new socket and listener";
+	   try {
+		    _sctp_4G_connection = std::make_unique<SctpConnection>(*req, _uplink_handler);
+		     } catch (...) {
+			      res->set_result(InitRes::INIT_FAIL);
+			      return Status::OK;
+		     }
+	    _sctp_4G_connection->Start();
```

Comment:
This logic can be abstracted into a create_sctp_connection(std::string) which can be called for both S1AP and NGAP.

Reply:
Hi Shruti, can we go with given approach now, later as enhancement we can change as per your comments?

Reply:
It will be good to clean this up in the current PR itself, it is a minor fix to separate into a new PR.